core/lib/dal/src/tee_proof_generation_dal.rs

```diff
@@ -1,60 +1,58 @@
 use std::time::Duration;
 
-use strum::{Display, EnumString};
 use zksync_db_connection::{
-    connection::Connection,
-    error::DalResult,
-    instrument::{InstrumentExt, Instrumented},
+    connection::Connection, error::DalResult, instrument::Instrumented,
     utils::pg_interval_from_duration,
 };
 use zksync_types::{tee_types::TeeType, L1BatchNumber};
 
-use crate::Core;
+use crate::{
+    models::storage_tee_proof::StorageTeeProof,
+    tee_verifier_input_producer_dal::TeeVerifierInputProducerJobStatus, Core,
+};
 
 #[derive(Debug)]
 pub struct TeeProofGenerationDal<'a, 'c> {
     pub(crate) storage: &'a mut Connection<'c, Core>,
 }
 
-#[derive(Debug, EnumString, Display)]
+#[derive(Debug, sqlx::Type)]
+#[sqlx(type_name = "tee_proofs_job_status")]
```

Comment:
While I appreciate the strive for strong typing, this looks like a somewhat dangerous change (e.g. things can go wrong with it). With dangerous changes, it's usually better to introduce them separately, so that if something goes wrong, you only have to rollback the _exact_ change and not everything else that was introduced with it.

Reply:
The TEE stuff is at a very early stage, so I wouldn't worry too much about the stability of its already deployed environments. However, you are absolutely right in general. I can split it into two PRs if you insist.

Reply:
Even if the component is not in active use, crashloop in any env would cause the on-call team to react and investigate. Depending on the change, a change in component A can cause other components to crash too. So it's always better to develop assuming that your feature can be rolled back, regardless of the readiness state.